example/client/cmd/util.go

```diff
@@ -50,23 +54,112 @@ func deleteAPIClient(cs *clientState) {
 	cs.connection = nil
 }
 
-func parseAttributes(attributes string) (map[string]string, error) {
-	attrs := make(map[string]string)
-	if len(attributes) > 0 {
-		for _, a := range strings.Split(attributes, ",") {
-			i := strings.Index(a, "=")
-			if i < 0 {
-				return nil, fmt.Errorf("Attribute value %v does not include an = sign", a)
-			} else if i == 0 {
-				return nil, fmt.Errorf("Attribute value %v does not contain a valid name", a)
+func parseAttributes(rootArgs *rootArgs) (*mixerpb.Attributes, error) {
+	attrs := mixerpb.Attributes{}
+	attrs.Dictionary = make(map[int32]string)
+
+	// once again, the following boilerplate would be more succinct with generics...
+
+	for i := 0; i < 6; i++ {
```

Comment:
can we do this with an unexported enum, so the cases are more readable?

and/or don't loop and just do each section separately (the shared bits are actually quite small -- the parsing of the args list, right?).